### ISSUE TITLE
Managed LU

### ISSUE THREAD

**Issue by cwhitten** [microsoft/BotFramework-Composer#690]
Breakdown of related work items

- [x] Users can author language understanding content inline for recognizedIntent event handler as well as any input action 
    - [x] S1: Add inline LU editor for recognizedIntent event 
    - [x] S1: Add inline LU editor for all input steps - number, datetime, text, choice, confirm
    - [x] S1: Serialization/ deserialization for dialog.lu file – open dev design: do we want separate files or a single .lu file? https://github.com/microsoft/BotFramework-Composer/issues/733
    - [ ] S1: Auto populate default name using LG scheme.
            - [ ] Follow up, separate ticket: Separate item for default naming for everything except Bot, Dialog, trigger (these 3 must have a name specified by user)
    - [ ] S2: Automatically add entities based on input step type (prebuilt number, datetime) 
        - [ ] ~list:choice~ for public preview. We need data model definition to do this correctly. 
        - [ ] ~list:confirm~ for public preview. We need data model definition to do this correctly. 
    - [ ] S3: Auto fill in/ suggest value property for input step based on LU – e.g. @number for numberInput etc. 
    - [ ] S3: Add a comment ‘recommended: please add at least 5 utterances message next to all LU input blocks’. Change to ‘more the examples, the better’ until user has at least 10 utterances per LU input block
- [ ] Once user provides examples and or language understanding constructs inline, contextually for inputs and recognizedIntent events, composer does the rest by automatically setting up required LUIS apps for their bot to function predictably
    - [ ] S1: Implement model suggestion library - cross training, sections collation.
    - [ ] Updates to .dialog files based on suggested models 
        - [ ] ~setting a score for intent events~ Current spec calls for 0.5 threshold. Let's re-evaluate this in the context of model evaluation + feedback
         - [ ] https://github.com/microsoft/botbuilder-dotnet/issues/2426
    - [ ] ~Model management library that orchestrates across ludown, dispatch down-sample, dispatch eval, lubuild.~ 
    - [ ] ~Update lubuild to work with down-sampled model definition~
    - [ ] ~Model down-sample library and wire up to down-sample~
    - [ ] ~Include model hierarchy as dispatch applications for LUIS~
- [x] Users immediately get to see any syntactic errors in the language understanding content
    - [x] Rewrite LU parser in ANTLR 
    - [x] S2: Re-wire LU validation rules with ANTLR parse tree 
    - [x] S3: Create a language server on top of Monco editor;   
    - [x] S3: Highlight syntax errors
  
## Task Track

| Item | Owner | Status
| - | - | -
| [X] UX Design for manged lu https://github.com/microsoft/BotFramework-Composer/issues/775 |Sangwoo | S12
| [ ] Form implementation for inline lu editor | Chris/Andy 
| [ ] Serialization/ deserialization for dialog.lu file https://github.com/microsoft/BotFramework-Composer/issues/733 | LG/Dong | S13, S14
| [ ] Implement model suggestion library - cross training, sections collation  https://github.com/microsoft/BotFramework-Composer/issues/805 | Dong | S13, S14
| [ ] Implement complicated sample bot  with managed LU https://github.com/microsoft/BotFramework-Composer/issues/806 | Lu | S13, S14

Open Items
- n/a
  
https://microsoft.sharepoint.com/:w:/r/teams/ConversationalAI785/_layouts/15/Doc.aspx?sourcedoc=%7BD4BE6F56-9089-45C2-8815-192E66786A0C%7D&file=Language%20Understanding%20in%20Bot%20Framework%20Designer.docx&action=default&mobileredirect=true

**Comment by boydc2014** [comment]
The first thing to unblock ourselves is
- [ ] bring the new lu toochain https://github.com/microsoft/BotFramework-Composer/issues/1556

**Comment by hibrenda** [comment]
Close this issue as most features are in R8.
